## Methyl-Seq Technology

### Short Introduction
Also referred to as bisulfite sequencing this method is used to study methylation patterns of DNA. Unmethylated cytosines are converted to uracil during library preparation and identified as thymine in the sequencing data. This can then be used to determine the percentage of methylated cytosines.

### Applications
Investigate the epigenome. This can play a role in a variety of areas such as cancer therapy, cloning, novel epigenetic biomarkers, or the gene environment interaction in general.

### Statistics
- Hierarchical Clustering based on Methyl-seq identified regions, with consideration of tissue type. 
- Hypergeometric test to identify differential methylation patterns.
- Mann-Whitney U-test to test whether the different fractions of methylations in different tissue types are statistically significant.

### References
-	https://www.illumina.com/techniques/sequencing/methylation-sequencing.html
-	https://eu.idtdna.com/pages/products/next-generation-sequencing/methods/methyl-seq
-	https://en.wikipedia.org/wiki/Bisulfite_sequencing
-	https://www.nature.com/articles/s41587-021-01108-x#Sec1
-	https://clinicalepigeneticsjournal.biomedcentral.com/articles/10.1186/s13148-019-0795-x

### Github names
Matthias-git-12  
HuandingJi  
YXC33  
XiaoqiZhangUZH
